client/src/components/ProfileForm.js

```diff
@@ -57,23 +137,52 @@ const TablePreferences = ({ preferences, handleCheckedBox }) => (
 			</Table.Row>
 		</Table.Header>
 		<Table.Body>
-			{preferences.map(preference => {
-				const { mi,km } = {...preference}
+			{preferences.map((preference, index ) => {
+				const name = 'user.preferences[' + index + '].value.'
+				const fieldName = {
+					'mi': name + 'mi',
+					'km': name + 'km'
+				}
 				return (
 					<Table.Row key={preference.name}>
-						<Table.Cell width={3}>{preference.name}</Table.Cell>
+						<Table.Cell width={3}>
+							{preference.name}
+						</Table.Cell>
 						<Table.Cell>	
-							<Checkbox label='Mi' checked={mi} onChange={handleCheckedBox}/>
-							<Checkbox label='KM' checked={km}/>
+							<Field name={fieldName.mi} component={Preference} type='checkbox' label="mi" />
+							<Field name={fieldName.km} component={Preference} type='checkbox' label="km" />
 						</Table.Cell>
 					</Table.Row>
 				)
 			})}
 		</Table.Body>
 	</Table>
 )
+const Preference = field => (
+	<Checkbox
+		{...field.input}
+		value={field.input.value ? 'on' : 'off'}
+		onChange={(e, { checked }) => field.input.onChange(checked)}
+		label={field.label} 		
```

Comment:
The km and mi selections don't work as expected. We shouldn't use a Checkbox in this case, as the user should either decide between km or mi. Try using Radio buttons instead.